src/components/event-types.ts

```diff
@@ -0,0 +1,59 @@
+export interface WeakEvent extends Record<string, unknown> {
+    // eslint-disable-next-line camelcase
+    event_id: string;
+    // eslint-disable-next-line camelcase
+    room_id: string;
+    sender: string;
+    content: Record<string, unknown>;
+    unsigned: {
+        age: number;
+    }
+    // eslint-disable-next-line camelcase
+    origin_server_ts: number;
+    // eslint-disable-next-line camelcase
+    state_key: string;
+    type: string;
+}
+
+export interface TypingEvent {
+    type: "m.typing";
+    content: {
+        // eslint-disable-next-line camelcase
+        user_ids: string[];
+    }
+    // eslint-disable-next-line camelcase
+    room_id: string;
+}
+
+export interface ReadReceiptEvent {
+    content: {
+        [eventId: string]: {
+            "m.read": {
+                [userId: string]: {
+                    ts: number
+                }
+            }
+        }
+    }
+    type: "m.receipt";
+    // eslint-disable-next-line camelcase
+    room_id: string;
+}
+
+export interface PresenceEvent {
+    content: {
```

Comment:
If the bridge shall be able to deal with home servers that are not spec-compliant, all properties would be marked optional and tested to have the right type (including the root object.

Reply:
>If the bridge shall be able to deal with home servers that are not spec-compliant

We don't plan to do that at all. Writing a bridge would be a headache if that were the case :D. I've merely copied https://matrix.org/docs/spec/client_server/r0.6.0#m-presence which only has one required field.